lib/log.js

```diff
@@ -74,9 +74,16 @@ Lug.prototype.activityEvent = function(event, uid, request) {
   if (request.headers['user-agent']) {
     info.userAgent = request.headers['user-agent']
   }
-  if (request.payload && request.payload.service) {
-    info.service = request.payload.service
+  try {
```

Comment:
This looks weird I know, but it seemed cleaner than explicitly checking both `request.payload` and `request.query` before dereferencing them.

Initially I assumed we could get rid of the checks entirely because Hapi always creates both objects. But a whole bunch of tests failed when I did that, hence catching the `ReferenceError` seemed like the next best thing to do.